File: sdks/python/seo_platform/resources/backlinks.py
"""Backlinks resource"""
from typing import Dict, Any, Optional


class Backlinks:
    """Backlinks resource - Monitor and analyze backlinks"""

    def __init__(self, client):
        self.client = client

    def list(
        self,
        project_id: str,
        status: Optional[str] = None,
    ) -> Dict[str, Any]:
        """
        List backlinks for a project

        Args:
            project_id: Project ID
            status: Filter by status ('active', 'lost', 'pending')

        Returns:
            Dictionary with backlinks data
        """
        params = {}
        if status:
            params['status'] = status

        return self.client.request(
            'GET',
            f'/projects/{project_id}/backlinks',
            params=params,
        )

    def stats(self, project_id: str) -> Dict[str, Any]:
        """
        Get backlink statistics

        Args:
            project_id: Project ID

        Returns:
            Backlink statistics
        """
        return self.client.request('GET', f'/projects/{project_id}/backlinks/stats')

    def refresh(self, project_id: str) -> None:
        """
        Refresh backlinks for a project

        Args:
            project_id: Project ID
        """
        self.client.request('POST', f'/projects/{project_id}/backlinks/refresh')

File: sdks/javascript/src/resources/backlinks.ts
import { AxiosInstance } from 'axios';

export interface Backlink {
  id: string;
  sourceUrl: string;
  targetUrl: string;
  anchorText?: string;
  status: 'active' | 'lost' | 'pending';
  domainAuthority?: number;
  pageAuthority?: number;
  isDoFollow: boolean;
  firstSeenAt: string;
  lastSeenAt: string;
  lostAt?: string;
}

export interface BacklinkStats {
  total: number;
  active: number;
  lost: number;
  doFollow: number;
  noFollow: number;
  averageDomainAuthority: number;
}

/**
 * Backlinks resource
 * Monitor and analyze backlinks
 */
export class Backlinks {
  constructor(private axios: AxiosInstance) {}

  /**
   * List backlinks for a project
   */
  async list(projectId: string, status?: string): Promise<{ data: Backlink[] }> {
    const response = await this.axios.get(`/projects/${projectId}/backlinks`, {
      params: { status },
    });
    return response.data;
  }

  /**
   * Get backlink statistics
   */
  async stats(projectId: string): Promise<BacklinkStats> {
    const response = await this.axios.get(`/projects/${projectId}/backlinks/stats`);
    return response.data;
  }

  /**
   * Refresh backlinks for a project
   */
  async refresh(projectId: string): Promise<void> {
    await this.axios.post(`/projects/${projectId}/backlinks/refresh`);
  }
}
